### ISSUE TITLE
[JENKINS-39801] Waiting for next available executor

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/kubernetes-plugin#1825]
<p>Using this plugin, the first job that starts instantly boots a new pod.</p>

<p>However, all other jobs that start <em>while a job is already running</em> first show the below two messages:</p>

<p>```<br/>
Still waiting to schedule task<br/>
Waiting for next available executor<br/>
```</p>

<p>Then, after a while, they eventually continue, and boot up a new pod as well.</p>

<p>How can I make all jobs instantly use new pods, instead of waiting for existing pods to become available?</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jeanmertz">jeanmertz</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801" target="_blank">Waiting for next available executor</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/csanchez.png" title="csanchez's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=csanchez">csanchez</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Minor
<li><b>component(s)</b>: kubernetes-plugin
<li><b>resolution</b>: Won't Do
<li><b>resolved</b>: 2018-05-25T09:21:36+00:00
<li><b>votes</b>: 2
<li><b>watchers</b>: 5
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Using this plugin, the first job that starts instantly boots a new pod.</p>

<p>However, all other jobs that start <em>while a job is already running</em> first show the below two messages:</p>

<p>```
Still waiting to schedule task
Waiting for next available executor
```</p>

<p>Then, after a while, they eventually continue, and boot up a new pod as well.</p>

<p>How can I make all jobs instantly use new pods, instead of waiting for existing pods to become available?</p></pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [Pasted image at 2017_01_04 17_10.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/35348/Pasted%20image%20at%202017_01_04%2017_10.png)
  > ![Pasted image at 2017_01_04 17_10.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/35348/Pasted%20image%20at%202017_01_04%2017_10.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>paul_symphony:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=277876&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-277876">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I am experiencing a similar, perhaps the same, issue.  I am seeing only a single job executing at a time.  The following is sample output that I see for each job &#8211; it repeats, as the jobs run serially instead of in parallel.</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Nov 20, 2016 3:58:54 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud provision
INFO: Excess workload after pending Spot instances: 1
Nov 20, 2016 3:58:54 PM hudson.slaves.NodeProvisioner$StandardStrategyImpl apply
INFO: Started provisioning Kubernetes Pod Template from local cluster with 1 executors. Remaining excess workload: 0
Nov 20, 2016 3:58:54 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud$ProvisioningCallback call
INFO: Created Pod: <span class="code-keyword">default</span>-6ee77ae7fd41
Nov 20, 2016 3:58:54 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud$ProvisioningCallback call
INFO: Waiting <span class="code-keyword">for</span> Pod to be scheduled (0/100): <span class="code-keyword">default</span>-6ee77ae7fd41
Nov 20, 2016 3:58:55 PM hudson.TcpSlaveAgentListener$ConnectionHandler run
INFO: Accepted connection #23 from /10.128.1.18:50036
Nov 20, 2016 3:58:55 PM hudson.TcpSlaveAgentListener$ConnectionHandler error
WARNING: Connection #23 is aborted: Disabled protocol:Protocol:JNLP3-connect
Nov 20, 2016 3:58:55 PM hudson.TcpSlaveAgentListener$ConnectionHandler run
INFO: Accepted connection #24 from /10.128.1.18:50038
Nov 20, 2016 3:59:01 PM hudson.node_monitors.ResponseTimeMonitor$1 monitor
WARNING: Making <span class="code-keyword">default</span>-6dcb9e264ecf offline because it’s not responding
Nov 20, 2016 3:59:04 PM hudson.slaves.NodeProvisioner$2 run
INFO: Kubernetes Pod Template provisioning successfully completed. We have now 3 computer(s)
Nov 20, 2016 3:59:07 PM hudson.model.Run execute
INFO: freestyle-test #10 main build action completed: SUCCESS
Nov 20, 2016 3:59:07 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
INFO: Terminating Kubernetes instance <span class="code-keyword">for</span> slave <span class="code-keyword">default</span>-6ee77ae7fd41
Nov 20, 2016 3:59:07 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
INFO: Terminated Kubernetes instance <span class="code-keyword">for</span> slave <span class="code-keyword">default</span>-6ee77ae7fd41
Nov 20, 2016 3:59:07 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
INFO: Disconnected computer <span class="code-keyword">default</span>-6ee77ae7fd41
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>I am experiencing a similar, perhaps the same, issue.  I am seeing only a single job executing at a time.  The following is sample output that I see for each job &#8211; it repeats, as the jobs run serially instead of in parallel.</p>

<pre>
Nov 20, 2016 3:58:54 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud provision
INFO: Excess workload after pending Spot instances: 1
Nov 20, 2016 3:58:54 PM hudson.slaves.NodeProvisioner$StandardStrategyImpl apply
INFO: Started provisioning Kubernetes Pod Template from local cluster with 1 executors. Remaining excess workload: 0
Nov 20, 2016 3:58:54 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud$ProvisioningCallback call
INFO: Created Pod: <span class="code-keyword">default</span>-6ee77ae7fd41
Nov 20, 2016 3:58:54 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesCloud$ProvisioningCallback call
INFO: Waiting <span class="code-keyword">for</span> Pod to be scheduled (0/100): <span class="code-keyword">default</span>-6ee77ae7fd41
Nov 20, 2016 3:58:55 PM hudson.TcpSlaveAgentListener$ConnectionHandler run
INFO: Accepted connection #23 from /10.128.1.18:50036
Nov 20, 2016 3:58:55 PM hudson.TcpSlaveAgentListener$ConnectionHandler error
WARNING: Connection #23 is aborted: Disabled protocol:Protocol:JNLP3-connect
Nov 20, 2016 3:58:55 PM hudson.TcpSlaveAgentListener$ConnectionHandler run
INFO: Accepted connection #24 from /10.128.1.18:50038
Nov 20, 2016 3:59:01 PM hudson.node_monitors.ResponseTimeMonitor$1 monitor
WARNING: Making <span class="code-keyword">default</span>-6dcb9e264ecf offline because it’s not responding
Nov 20, 2016 3:59:04 PM hudson.slaves.NodeProvisioner$2 run
INFO: Kubernetes Pod Template provisioning successfully completed. We have now 3 computer(s)
Nov 20, 2016 3:59:07 PM hudson.model.Run execute
INFO: freestyle-test #10 main build action completed: SUCCESS
Nov 20, 2016 3:59:07 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
INFO: Terminating Kubernetes instance <span class="code-keyword">for</span> slave <span class="code-keyword">default</span>-6ee77ae7fd41
Nov 20, 2016 3:59:07 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
INFO: Terminated Kubernetes instance <span class="code-keyword">for</span> slave <span class="code-keyword">default</span>-6ee77ae7fd41
Nov 20, 2016 3:59:07 PM org.csanchez.jenkins.plugins.kubernetes.KubernetesSlave _terminate
INFO: Disconnected computer <span class="code-keyword">default</span>-6ee77ae7fd41
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/csanchez.png" title="csanchez's avatar" /> csanchez:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=278070&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-278070">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>You are probably hitting this <a href="https://github.com/jenkinsci/mesos-plugin#over-provisioning-flags" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/mesos-plugin#over-provisioning-flags</a></p>

<blockquote>
<p>By default, Jenkins spawns slaves conservatively. Say, if there are 2 builds in queue, it won't spawn 2 executors immediately. It will spawn one executor and wait for sometime for the first executor to be freed before deciding to spawn the second executor. Jenkins makes sure every executor it spawns is utilized to the maximum. If you want to override this behaviour and spawn an executor for each build in queue immediately without waiting, you can use these flags during Jenkins startup: -Dhudson.slaves.NodeProvisioner.MARGIN=50 -Dhudson.slaves.NodeProvisioner.MARGIN0=0.85</p></blockquote>
</pre>
</details>

</ul>
</details>

<p>You are probably hitting this <a href="https://github.com/jenkinsci/mesos-plugin#over-provisioning-flags" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/mesos-plugin#over-provisioning-flags</a></p>

<blockquote>
<p>By default, Jenkins spawns slaves conservatively. Say, if there are 2 builds in queue, it won't spawn 2 executors immediately. It will spawn one executor and wait for sometime for the first executor to be freed before deciding to spawn the second executor. Jenkins makes sure every executor it spawns is utilized to the maximum. If you want to override this behaviour and spawn an executor for each build in queue immediately without waiting, you can use these flags during Jenkins startup: -Dhudson.slaves.NodeProvisioner.MARGIN=50 -Dhudson.slaves.NodeProvisioner.MARGIN0=0.85</p></blockquote>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jeanmertz:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=278891&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-278891">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes, you are correct. That does seem to be the issue.</p>

<p>Perhaps this should be part of the plugin documentation?</p>
</pre>
</details>

</ul>
</details>

<p>Yes, you are correct. That does seem to be the issue.</p>

<p>Perhaps this should be part of the plugin documentation?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/csanchez.png" title="csanchez's avatar" /> csanchez:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=279195&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-279195">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Added to README in <a href="https://github.com/jenkinsci/kubernetes-plugin/pull/98" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/kubernetes-plugin/pull/98</a></p>
</pre>
</details>

</ul>
</details>

<p>Added to README in <a href="https://github.com/jenkinsci/kubernetes-plugin/pull/98" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/kubernetes-plugin/pull/98</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jeanmertz:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=282051&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-282051">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Starting Jenkins with the config provided in the GitHub repository, and these flags:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>-Dhudson.slaves.NodeProvisioner.MARGIN=50 -Dhudson.slaves.NodeProvisioner.MARGIN0=0.85
</pre>
</div></div>

<p>And using a simple Pipeline script like this:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>podTemplate(label: 'mypod', containers: [
    containerTemplate(name: 'debian', image: 'debian', ttyEnabled: true, command: 'cat'),
    containerTemplate(name: 'ubuntu', image: 'ubuntu', ttyEnabled: true, command: 'cat')
  ]) {
    node ('mypod') {
        stage('stage 1') {
            container('debian') {
                echo 'stage 1'
                sleep 30
            }
        }

        stage('stage 2') {
            container('ubuntu') {
                echo 'stage 2'
                sleep 30
            }
        }
    }
}
</pre>
</div></div>

<p>And a <tt>containerCap</tt> of 1000+</p>

<p>I still see jobs being started in a stacked way, when starting 10+ jobs, the last job waits more several minutes before it actually starts running (see attached screenshot)</p>
</pre>
</details>

</ul>
</details>

<p>Starting Jenkins with the config provided in the GitHub repository, and these flags:</p>



```
-Dhudson.slaves.NodeProvisioner.MARGIN=50 -Dhudson.slaves.NodeProvisioner.MARGIN0=0.85

```

<p>And using a simple Pipeline script like this:</p>



```
podTemplate(label: 'mypod', containers: [
    containerTemplate(name: 'debian', image: 'debian', ttyEnabled: true, command: 'cat'),
    containerTemplate(name: 'ubuntu', image: 'ubuntu', ttyEnabled: true, command: 'cat')
  ]) {
    node ('mypod') {
stage('stage 1') {
    container('debian') {
echo 'stage 1'
sleep 30
    }
}

stage('stage 2') {
    container('ubuntu') {
echo 'stage 2'
sleep 30
    }
}
    }
}

```

<p>And a <tt>containerCap</tt> of 1000+</p>

<p>I still see jobs being started in a stacked way, when starting 10+ jobs, the last job waits more several minutes before it actually starts running (see attached screenshot)</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/lionelve.png" title="lionelve's avatar" /> lionelve:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=286883&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-286883">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This becomes more apparent when you try to run pods in parallel.</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">podTemplate(name: <span class="code-quote">'jenkins-slave'</span>, label: <span class="code-quote">'kube-node'</span>,  instanceCap: 4,
		containers: [
			containerTemplate(
				name: <span class="code-quote">'maven'</span>, 
				image: <span class="code-quote">"${registry}:5000/jenkins/slave-maven"</span>,
				command: <span class="code-quote">'/bin/bash'</span>,
				args: '',
				alwaysPullImage: <span class="code-keyword">true</span>,
				ttyEnabled: <span class="code-keyword">true</span>)
		])
	{
 parallel (
		    <span class="code-quote">"Unit"</span>: {
		                node(<span class="code-quote">'kube-node'</span>) {
					            container(<span class="code-quote">'maven'</span>) {
                                                      sleep(30)
			                }
		                }
		    },
		    <span class="code-quote">"Integration"</span>: {
		                    node(<span class="code-quote">'kube-node'</span>) {
				                    container(<span class="code-quote">'maven'</span>) {
                                                      sleep(30)
			                    }
		                    }
		    }
		)
}
</pre>
</div></div>

<p>The second pod only gets created after the first on finishes effectively serialising the parallel steps.</p>
</pre>
</details>

</ul>
</details>

<p>This becomes more apparent when you try to run pods in parallel.</p>


<pre>
podTemplate(name: <span class="code-quote">'jenkins-slave'</span>, label: <span class="code-quote">'kube-node'</span>,  instanceCap: 4,
		containers: [
			containerTemplate(
				name: <span class="code-quote">'maven'</span>, 
				image: <span class="code-quote">"${registry}:5000/jenkins/slave-maven"</span>,
				command: <span class="code-quote">'/bin/bash'</span>,
				args: '',
				alwaysPullImage: <span class="code-keyword">true</span>,
				ttyEnabled: <span class="code-keyword">true</span>)
		])
	{
 parallel (
		    <span class="code-quote">"Unit"</span>: {
		node(<span class="code-quote">'kube-node'</span>) {
					    container(<span class="code-quote">'maven'</span>) {
      sleep(30)
			}
		}
		    },
		    <span class="code-quote">"Integration"</span>: {
		    node(<span class="code-quote">'kube-node'</span>) {
				    container(<span class="code-quote">'maven'</span>) {
      sleep(30)
			    }
		    }
		    }
		)
}
</pre>

<p>The second pod only gets created after the first on finishes effectively serialising the parallel steps.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/lionelve.png" title="lionelve's avatar" /> lionelve:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=287079&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-287079">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm running Jenkins in a container and was not setting the NodeProvisioner properties correctly.</p>

<p>After fixing that the pods are created in parallel. It takes a little while for the second one to be created but they do run simultaneously. </p>
</pre>
</details>

</ul>
</details>

<p>I'm running Jenkins in a container and was not setting the NodeProvisioner properties correctly.</p>

<p>After fixing that the pods are created in parallel. It takes a little while for the second one to be created but they do run simultaneously. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/lionelve.png" title="lionelve's avatar" /> lionelve:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=287715&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-287715">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jeanmertz" class="user-hover" rel="jeanmertz">jeanmertz</a> In your case try generating a random label and use that in place of  'mypod' (both in the podTemplate and the node)</p>

<p>Something like </p>


<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-keyword">import</span> <span class="code-keyword">static</span> java.util.UUID.randomUUID
def uuid = randomUUID() as <span class="code-object">String</span>
def label = uuid.take(8)

podTemplate(label: <span class="code-quote">"pod-${label}"</span>, instanceCap: 1,
			  containers: ...)
		 {
			node(<span class="code-quote">"pod-${label}"</span>){
				 ...
			}
		}
}
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jeanmertz" class="user-hover" rel="jeanmertz">jeanmertz</a> In your case try generating a random label and use that in place of  'mypod' (both in the podTemplate and the node)</p>

<p>Something like </p>



<pre>
<span class="code-keyword">import</span> <span class="code-keyword">static</span> java.util.UUID.randomUUID
def uuid = randomUUID() as <span class="code-object">String</span>
def label = uuid.take(8)

podTemplate(label: <span class="code-quote">"pod-${label}"</span>, instanceCap: 1,
			  containers: ...)
		 {
			node(<span class="code-quote">"pod-${label}"</span>){
				 ...
			}
		}
}
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/lionelve.png" title="lionelve's avatar" /> lionelve:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=288625&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-288625">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>After much head-banging over this I finally got pods created concurrently and in parallel. </p>

<ol>
	<li>Using a different (random) label for each podTemplate does create pods in parallel but the plugin treats all podTemplates in a Run as nested templates even when they are not.</li>
	<li>The hacky workaround to stop this is to clear all previously stacked template names before creating a new one. Call this function before any occurrence of podTemplate:</li>
</ol>


<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">@NonCPS
def clearTemplateNames() {
	def r = script.currentBuild.rawBuild
	def action = r.getAction(PodTemplateAction.class);
	<span class="code-keyword">if</span>(action) {
		action.names.clear()
	}
}
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>After much head-banging over this I finally got pods created concurrently and in parallel. </p>

<ol>
	<li>Using a different (random) label for each podTemplate does create pods in parallel but the plugin treats all podTemplates in a Run as nested templates even when they are not.</li>
	<li>The hacky workaround to stop this is to clear all previously stacked template names before creating a new one. Call this function before any occurrence of podTemplate:</li>
</ol>



<pre>
@​NonCPS
def clearTemplateNames() {
	def r = script.currentBuild.rawBuild
	def action = r.getAction(PodTemplateAction.class);
	<span class="code-keyword">if</span>(action) {
		action.names.clear()
	}
}
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/csanchez.png" title="csanchez's avatar" /> csanchez:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=302592&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-302592">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The issue with nesting podTemplates is <a href="https://issues.jenkins.io/browse/JENKINS-42184" title="Unable to run pods in parallel due to template nesting" class="issue-link" data-issue-key="JENKINS-42184"><del>JENKINS-42184</del></a></p>
</pre>
</details>

</ul>
</details>

<p>The issue with nesting podTemplates is <a href="https://issues.jenkins.io/browse/JENKINS-42184" title="Unable to run pods in parallel due to template nesting" class="issue-link" data-issue-key="JENKINS-42184"><del>JENKINS-42184</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>jeanmertz:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=302598&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-302598">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>sorry for not replying earlier. using randomised labels did solve the issue for us <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/thumbs_up.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>sorry for not replying earlier. using randomised labels did solve the issue for us <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/thumbs_up.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>pastushenko:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-39801?focusedId=343770&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-343770">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hello Guys!</p>

<p>I use kubernetes plugin for running freestyle projects.</p>

<p>I configured default pod template and labeled it, for example "k8s-runner".</p>

<p>I configured a freestyle project and restricted to run this project in "k8s-runner".</p>

<p>I have same issue. The job wait ~90-120 seconds if there is already running jobs (there are existing pods).</p>

<p>If there are no running jobs - job will start in 20-30 seconds.</p>

<p>The workaround from previous comments works only for pipelines.</p>

<p>Is there any solution for freestyle projects?</p>

<p> </p>

<p>Tanks.</p>
</pre>
</details>

</ul>
</details>

<p>Hello Guys!</p>

<p>I use kubernetes plugin for running freestyle projects.</p>

<p>I configured default pod template and labeled it, for example "k8s-runner".</p>

<p>I configured a freestyle project and restricted to run this project in "k8s-runner".</p>

<p>I have same issue. The job wait ~90-120 seconds if there is already running jobs (there are existing pods).</p>

<p>If there are no running jobs - job will start in 20-30 seconds.</p>

<p>The workaround from previous comments works only for pipelines.</p>

<p>Is there any solution for freestyle projects?</p>

<p> </p>

<p>Tanks.</p>
